draft-ietf-avtcore-cc-feedback-message-01.txt

```diff
@@ -0,0 +1,616 @@
+   described in Section 4 of [RFC3550].
+
+   RTCP congestion control feedback packets SHOULD include a report
+   block for each SSRC that is being congestion controlled.  The
+   sequence number ranges reported on in consecutive reports for an SSRC
+   SHOULD be consecutive and SHOULD NOT overlap (i.e., begin_seq for a
+   report is expected to be one greater, modulo 65535, than end_seq of
+   the previous report for that SSRC).  If overlapping reports are sent,
+   the information in the later report updates that in any previous
+   reports for packets included in both reports (although note that such
+   updated information will likely arrive too late to affect congestion
+   control decisions at the sender).  Reports that cover RTP sequence
+   number ranges that are more than 16384 (i.e., one quarter of the
+   sequence number space) ahead of the last end_seq received from an
+   SSRC, or behind the last begin_seq received from an SSRC, modulo
+   65535 to account for wrap-around, MUST be ignored.
+
+   If no packets are received from an SSRC in a reporting interval, then
+   no report block is sent for that SSRC.  A regular SR/RR packet SHOULD
+   be sent instead, since the non-increased extended highest sequence
+   number received field of that SR/RR packet will inform the sender
+   that no packets have been received.
+
+4.  Feedback Frequency and Overhead
+
+   There is a trade-off between speed and accuracy of reporting, and the
+   overhead of the reports.  [I-D.ietf-rmcat-rtp-cc-feedback] discusses
+   this trade-off, suggests desirable RTCP feedback rates, and provides
+   guidance on how to configure the RTCP bandwidth fraction, etc., to
+   make appropriate use of the reporting block described in this memo.
+   Specifications for RTP congestion control algorithms can also provide
+   guidance.
+
+   It is a general understanding that the congestion control algorithms
+   will work better with more frequent feedback - per packet feedback.
+   However, RTCP bandwidth and transmission rules put some upper limits
+   on how frequently the RTCP feedback messages can be send from the RTP
+   receiver to the RTP sender.  It has been shown
+   [I-D.ietf-rmcat-rtp-cc-feedback] that in most cases a per frame
+   feedback is a reasonable assumption on how frequent the RTCP feedback
+   messages can be transmitted.  It has also been noted that even if a
+   higher frequency of feedback is desired it is not viable if the
+
+
+
+Sarker, et al.          Expires September 3, 2018               [Page 6]
+
+Internet-Draft     Congestion Control Feedback in RTCP        March 2018
+
+
+   feedback messages starts to compete against the RTP traffic on the
+   feedback path during congestion period.  Analyzing the feedback
+   interval requirement [feedback-requirements] it can be seen that the
+   candidate algorithms can perform with a feedback interval range of
+   50-200ms.  A value within this range need to be negotiated at session
+   setup.
```

Comment:
Are we adding a SDP consideration for this?

I would lean that this value use SDP declarative style and not be negotiated, i.e., if Sender A wants 50ms and Sender B wants 200ms, it be 50 and 200ms respectively for Receiver B and A, and not have to pick a lower or higher value...

Reply:
Agree, although it's the usual RTCP BW parameters, I think. To be specified in the feedback-requirements draft.